python/python/glide/async_commands/redis_modules/search/field.py

```diff
@@ -0,0 +1,375 @@
+            args.extend(["AS", self.alias])
+        return args
+
+
+class TextField(Field):
+    """
+    Class for defining text fields in a schema.
+
+    Args:
+        name (str): The name of the text field.
+        alias (Optional[str]): An alias for the field.
+        sortable (Optional[SortableOption]): Whether the field is sortable, options are: SORTABLE and SORTABLE_UNF.
+        no_stem (bool): Whether to disables stemming when indexing field values.
+        no_index (bool): Indicate whether the field should be excluded from indexing.
+        phonetic (Optional[PhoneticType]): Activate phonetic matching for text searches, specifies the phonetic algorithm and language used (e.g., English: dm:en).
+        weight (Optional[float]): Specify attribute importance of this attribute when calculating result accuracy.
+        with_suffix_trie (bool): Improve query performance by utilizing a suffix trie for contains (foo) and suffix (*foo) queries.
+            This feature avoids brute-force searches on the trie.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        alias: Optional[str] = None,
+        sortable: Optional[SortableOption] = None,
+        no_stem: bool = False,
+        no_index: bool = False,
+        phonetic: Optional[PhoneticType] = None,
+        weight: Optional[float] = None,
+        with_suffix_trie: bool = False,
+    ):
+        """Initialize a new TextField instance."""
+        super().__init__(name, alias)
+        self.sortable = sortable
+        self.no_stem = no_stem
+        self.no_index = no_index
+        self.phonetic = phonetic
+        self.weight = weight
+        self.with_suffix_trie = with_suffix_trie
+
+    def get_field_args(self) -> List[str]:
+        """
+        Get the arguments representing the text field.
+
+        Returns:
+            List[str]: A list of text field arguments.
+        """
+        args = super().get_field_args()
+        args.append("TEXT")
+        if not self.sortable and self.no_index:
+            raise ValueError("Cannot be both non-storable and non-indexed.")
+        if self.sortable:
+            args.extend(self.sortable.value.split(" "))
+        if self.no_stem:
+            args.append("NOSTEM")
+        if self.phonetic:
+            args.extend(["PHONETIC", self.phonetic.value])
+        if self.weight:
+            args.extend(["WEIGHT", str(self.weight)])
+        if self.with_suffix_trie:
+            args.append("WITHSUFFIXTRIE")
+        if self.no_index:
+            args.append("NOINDEX")
+        return args
+
+
+class TagField(Field):
+    """
+    Class for defining tag fields in a schema.
+
+    Args:
+        name (str): The name of the tag field.
+        alias (Optional[str]): An alias for the field.
+        sortable (Optional[SortableOption]): Whether the field is sortable, options are: SORTABLE and SORTABLE_UNF.
+        no_index (bool): Indicate whether the field should be excluded from indexing.
```

Comment:
> ... If an attribute has NOINDEX and doesn't have SORTABLE, it will just be ignored by the index.

we need or to document that it will be ignored or to raise an excption if no_index is passed without sortable. please make sure if there is a meaning of no_index without sortable and think what options we have to make the API easier for the user